### PR TITLE
 OCPBUGS-59637: add k8s-cacert to hybrid overlay cmd

### PR DESCRIPTION
**Pull request by jrvaldes**
The hybrid-overlay service now includes the --k8s-cacert flag pointing to the trusted CA bundle path to ensure proper certificate validation.

Additionally, comprehensive unit tests have been added for the hybridOverlayConfiguration function covering all parameter combinations, edge cases, command structure validation, and service property consistency.

The tests validate:
- Basic configuration scenarios (minimal, VXLAN port, debug logging)
- Edge cases (zero port, whitespace, special characters)
- Command structure and argument ordering
- Service definition properties
- Parameter independence and flag uniqueness

Fixes OCPBUGS-59637

before:
```
I0812 14:19:07.202384    2340 certificate_manager.go:356] kubernetes.io/kube-apiserver-client: Rotating certificates
E0812 14:19:07.205049    2340 certificate_manager.go:562] "Unhandled Error" err="kubernetes.io/kube-apiserver-client: Failed while requesting a signed certificate from the control plane: cannot create certificate signing request: Post \"http://localhost:8443/apis/certificates.k8s.io/v1/certificatesigningrequests\": dial tcp 127.0.0.1:8443: connectex: No connection could be made because the target machine actively refused it." logger="UnhandledError"
E0812 14:19:07.205049    2340 certificate_manager.go:562] "Unhandled Error" err="kubernetes.io/kube-apiserver-client: Failed while requesting a signed certificate from the control plane: cannot create certificate signing request: Post \"http://localhost:8443/apis/certificates.k8s.io/v1/certificatesigningrequests\": dial tcp 127.0.0.1:8443: connectex: No connection could be made because the target machine actively refused it." logger="UnhandledError"
E0812 14:19:07.205049    2340 certificate_manager.go:562] "Unhandled Error" err="kubernetes.io/kube-apiserver-client: Failed while requesting a signed certificate from the control plane: cannot create certificate signing request: Post \"http://localhost:8443/apis/certificates.k8s.io/v1/certificatesigningrequests\": dial tcp 127.0.0.1:8443: connectex: No connection could be made because the target machine actively refused it." logger="UnhandledError"

```

now:
```
I0812 19:46:00.409816    4420 certificate_manager.go:356] kubernetes.io/kube-apiserver-client: Certificate rotation is enabled
I0812 19:46:00.410361    4420 certificate_manager.go:356] kubernetes.io/kube-apiserver-client: Rotating certificates
I0812 19:46:00.410361    4420 kube.go:419] Waiting for certificate
I0812 19:46:00.411987    4420 cert_rotation.go:140] Starting client certificate rotation controller
I0812 19:46:00.411987    4420 envvar.go:172] "Feature gate default state" feature="ClientsAllowCBOR" enabled=false
I0812 19:46:00.411987    4420 envvar.go:172] "Feature gate default state" feature="ClientsPreferCBOR" enabled=false
I0812 19:46:00.413071    4420 envvar.go:172] "Feature gate default state" feature="InformerResourceVersion" enabled=false
I0812 19:46:00.413071    4420 envvar.go:172] "Feature gate default state" feature="WatchListClient" enabled=false
I0812 19:46:00.431169    4420 reflector.go:313] Starting reflector *v1.CertificateSigningRequest (0s) from k8s.io/client-go/tools/watch/informerwatcher.go:146
I0812 19:46:00.431169    4420 reflector.go:349] Listing and watching *v1.CertificateSigningRequest from k8s.io/client-go/tools/watch/informerwatcher.go:146
I0812 19:46:00.432586    4420 reflector.go:376] Caches populated for *v1.CertificateSigningRequest from k8s.io/client-go/tools/watch/informerwatcher.go:146
I0812 19:46:00.433604    4420 csr.go:261] certificate signing request csr-znjhr is approved, waiting to be issued
I0812 19:46:00.440700    4420 csr.go:257] certificate signing request csr-znjhr is issued
I0812 19:46:00.440700    4420 reflector.go:319] Stopping reflector *v1.CertificateSigningRequest (0s) from k8s.io/client-go/tools/watch/informerwatcher.go:146
I0812 19:46:01.410921    4420 kube.go:426] Certificate found
I0812 19:46:01.411437    4420 cert_rotation.go:140] Starting client certificate rotation controller
I0812 19:46:01.449843    4420 certificate_manager.go:356] kubernetes.io/kube-apiserver-client: Certificate expiration is 2025-08-13 19:41:00 +0000 UTC, rotation deadline is 2025-08-13 16:16:02.548679651 +0000 UTC
I0812 19:46:01.449843    4420 certificate_manager.go:356] kubernetes.io/kube-apiserver-client: Waiting 20h30m1.098835951s for next certificate rotation

```

### CODE REVIEW

File: pkg/services/services_test.go [hunk 1]
```diff
@@ -41,3 +46,544 @@ func TestGetHostnameCmd(t *testing.T) {
 		})
 	}
 }
+
+func TestHybridOverlayConfiguration(t *testing.T) {
```
Comment (sebsoto) [inline pkg/services/services_test.go:50]: This PR is adding 550 lines of testing to test two potential string appends.
Please think carefully about this.
What value do these tests really add compared to the amount of  time it takes for people to review them, and the effort to change them when a new change needs to be made to the hybrid overlay configuration?

File: pkg/services/services_test.go [hunk 2]
```diff
@@ -41,3 +46,544 @@ func TestGetHostnameCmd(t *testing.T) {
...
+				// Verify it appears exactly once
+				count := strings.Count(result.Command, "--k8s-cacert")
+				assert.Equal(t, 1, count,
+					"k8s-cacert flag should appear exactly once")
+			})
+		}
+	})
+}
+
+func TestHybridOverlayConfigurationVXLANPortHandling(t *testing.T) {
```
Comment (sebsoto) [inline pkg/services/services_test.go:318]: What is this for?

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: pkg/services/services_test.go (hunk 2 above)
Comment (jrvaldes) [reply pkg/services/services_test.go]: removed

File: pkg/services/services.go [hunk 3]
```diff
@@ -108,6 +108,10 @@ func hybridOverlayConfiguration(vxlanPort string, debug bool) servicescm.Service
 	hybridOverlayServiceCmd := fmt.Sprintf("%s --node NODE_NAME --bootstrap-kubeconfig=%s --cert-dir=%s --cert-duration=24h "+
 		"--windows-service --logfile "+"%s\\hybrid-overlay.log", windows.HybridOverlayPath, windows.KubeconfigPath, windows.CniConfDir,
 		windows.HybridOverlayLogDir)
+
+	// append cacert option pointing to the trusted CA bundle path
+	hybridOverlayServiceCmd = fmt.Sprintf("%s --k8s-cacert %s", hybridOverlayServiceCmd, windows.TrustedCABundlePath)
```
[the remark this answers is not in the document]
Comment (jrvaldes) [reply pkg/services/services.go:113]: oh wait, it uses the k8s client /cert_rotation.go

```
I0812 22:16:29.810837    2560 cert_rotation.go:140] Starting client certificate rotation controller
```
[see logs here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_windows-machine-config-operator/3159/pull-ci-openshift-windows-machine-config-operator-master-aws-e2e-operator/1955368879337246720/artifacts/aws-e2e-operator/windows-e2e-operator-test/artifacts/nodes/ip-10-0-10-1.us-east-2.compute.internal/hybrid-overlay/hybrid-overlay.log)


So,  yes. The dynamicClientCert.loadClientCert() supports dynamic reloading of client certificates. The function checks for changes in the cert data and, if a change is detected, updates the in-memory certificate and triggers connection rotations.

https://github.com/openshift/kubernetes/blob/master/staging/src/k8s.io/client-go/transport/cert_rotation.go#L92

**Comment by jrvaldes** [comment]
@openshift/openshift-team-windows-containers  PTAL
